### PR TITLE
update armature

### PR DESCRIPTION
**Pull request by 2youyou2**
1.improve performance
2.add gotoAndPlay method
3.fix asynchronous reading error
4.change CCArmature::boundingBox to apply CCNode::boundingBox
5.fix different armatures may play the same movement on some android device

### CODE REVIEW

File: extensions/CocoStudio/Armature/CCBone.cpp [hunk 1]
```diff
@@ -74,6 +74,8 @@ CCBone *CCBone::create(const char *name)
     m_bBoneTransformDirty = true;
     m_eBlendType = BLEND_NORMAL;
     m_tWorldInfo = NULL;
+    m_pArmatureParentBone = NULL;
+    m_fDataVersion = 0;
```
Comment (xingsenma) [inline extensions/CocoStudio/Armature/CCBone.cpp:78]: Suggest dataVersion is used  as a global variables.

File: extensions/CocoStudio/Armature/utils/CCSpriteFrameCacheHelper.h [hunk 2]
```diff
@@ -47,14 +47,11 @@ class  CCSpriteFrameCacheHelper
      */
     void addSpriteFrameFromFile(const char *plistPath, const char *imagePath);
 
-    CCTextureAtlas *getTexureAtlasWithTexture(CCTexture2D *texture);
```
Comment (xingsenma) [inline extensions/CocoStudio/Armature/utils/CCSpriteFrameCacheHelper.h:50]: I think "CCSpriteFrameCacheHelper.h" this file can be delete from armature.

File: extensions/CocoStudio/Armature/CCBone.cpp (hunk 1 above)
Comment (2youyou2) [inline extensions/CocoStudio/Armature/CCBone.cpp:78]: There is a global dataVersion for current CCArmature's version
This one is the data version of the CocoStudio Editor.

File: extensions/CocoStudio/Armature/utils/CCSpriteFrameCacheHelper.h (hunk 2 above)
Comment (2youyou2) [inline extensions/CocoStudio/Armature/utils/CCSpriteFrameCacheHelper.h:50]: Yes, this class is useless now.

**Pull request merged**